.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,5 +1,18 @@
 <!--  Thanks for sending a pull request! -->
 
+#### Developer Checklist:
+<!--  Before requesting for code review, please ensure that the following tasks are completed. -->
```

Comment:
```suggestion
<!--  
Merging into the main branch implies your code is ready for production. Before requesting for code review, 
please ensure that the following tasks are completed. Otherwise, keep the PR drafted.
-->
```